diff -aur vanilla/ChibiOS_21.11.2/os/common/ports/AVR/compilers/GCC/rules.mk ChibiOS_21.11.2/os/common/ports/AVR/compilers/GCC/rules.mk
--- a/os/common/ports/AVR/compilers/GCC/rules.mk	2017-09-16 16:02:09.000000000 +0200
+++ b/os/common/ports/AVR/compilers/GCC/rules.mk	2022-08-25 20:21:42.431618728 +0200
@@ -81,6 +81,12 @@
 CFLAGS  += -std=gnu11 -mrelax -fdata-sections -ffunction-sections
 CFLAGS  += -Wundef -MMD -MP #-MF
 
+CPPFLAGS = $(MCFLAGS) -I. -gdwarf-2 $(CDEFS) $(OPT) -funsigned-char
+CPPFLAGS  += -funsigned-bitfields -fpack-struct -fshort-enums $(CWARN)
+CPPFLAGS  += -Wa,-adhlns=$(LSTDIR)/$(notdir $(<:.c=.lst)) $(DEFS)
+CPPFLAGS  += -std=gnu++11 -mrelax -fdata-sections -ffunction-sections
+CPPFLAGS  += -Wundef -MMD -MP #-MF
+
 #---------------- Assembler Options ----------------
 #  -Wa,...:   tell GCC to pass this to the assembler.
 #  -adhlns:   create listing
@@ -264,7 +270,7 @@
 lib: $(OBJS) $(BUILDDIR)/lib$(PROJECT).a
 
 $(BUILDDIR)/lib$(PROJECT).a: $(OBJS)
-	@$(AR) -r $@ $^
+	@$(AR) $@ $^
 	@echo
 	@echo Done
 
